### DISCUSSION TITLE
Using triggers over multiple projects

### DISCUSSION THREAD

**Discussion by aidan-rypens** [koenbeuk/EntityFrameworkCore.Triggered#132] [Q&A] [answered]
Is it possible to use a trigger over projects?

For example:
Project A (contains the DbContext)
Project B (uses a DbContext, another instance of it, but lives in a seperate project)

If we update an entity in project A, is it possible to fire a trigger that lives sololy in project B?
So the trigger lives in project B, and is passed in the Startup when creating the context.

To my understanding, both instantiate a different instance of the dbcontext and therefore updates to the dbcontext will only fire a trigger that is hooked on the same db context.

Just trying to understand a bit what is possible and how this could fit over projects. Am I right that this use case is not viable?

Thanks

**Comment by koenbeuk** [comment, marked as answer]
Any trigger regardless of where its code lives (ProjectA, ProjectB, a nuget package or even something dynamic) will be invoked when performing a Save operation for as long as those triggers are registered with the DbContext type ([or with your current ServiceProvider](https://github.com/koenbeuk/EntityFrameworkCore.Triggered#trigger-discovery)).

I'm not sure I understand this question so please feel free to clarify if the above does not help :)

**Reply by joshuasir** [reply]
Sorry i want to make sure

What if the Project A the event publisher that should trigger the trigger is in a separate app using different DBContext instance than Project B (but use the same dbcontext type from the same library provider) without EntityFrameworkCore.Triggered configuration in the startup. 

Where the EntityFrameworkCore.Triggered is only installed and initialized in the Project B, acting as the listener of the event and as the trigger executor.

Is it possible for the Project B to discover Project A event to execute the trigger?
